runtime/near-evm-runner/src/lib.rs

```diff
@@ -431,10 +590,18 @@ pub fn run_evm(
         storage_usage,
         prepaid_gas,
         is_view,
+        evm_gas,
```

Comment:
hm, given there is also prepaid_gas already passed that is used for things like transfers and promises - I'm not sure counting them separately is correct.

Given you already have EvmGasCounter -> it should receive prepaid NEAR gas, and then should support both charging evm gas and NEAR gas and returning left NEAR/EVM gas.

Reply:
The functionality is there, but not encapsulate in EvmGasCounter:
```

Reply:
Found it's very hard to move it EvmGasCounter, since EvmGasCounter needs to know whether it's deploy_code/call_function or other to determine max evm gas from given prepaid near gas. So EvmContext has to record whether it's a deploy_code/call_function context. Then if use a deploy_code context for call function, EvmGasCounter will have an incorrect max evm gas. This won't happen in prod because new EvmContext is created freshly on run_evm. But it happens in nearly all tests